packages/next/src/server/dev/next-dev-server.ts

```diff
@@ -268,7 +268,9 @@ export default class DevServer extends Server {
     await super.prepareImpl()
     await this.runInstrumentationHookIfAvailable()
     await this.matchers.reload()
-    this.setDevReady!()
+
+    this.ready?.resolve()
+    this.ready = undefined
```

Comment:
Any reason this is set to undefined?

Reply:
Because we didn't want to permit resolving more than once or rejecting after the value was resolved, this setting it to `undefined` was to act as a signal that the "value isn't of use anymore". Ideally, we'd want to do `if (this.ready) await this.ready.promise` to improve the performance, as it would allow us to avoid the `await` and subsequent yielding to the event loop.